lib/web_ui/lib/src/engine/view_embedder/flutter_view_manager.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine.dart';
+import 'package:ui/src/engine/js_interop/js_app.dart';
+
+/// Encapsulates the `viewData` Map, and automates the creation of `viewId`s.
+class FlutterViewManager {
+  /// Creates a FlutterViewManager with a [initialViewId].
+  ///
+  /// This defaults to [kImplicitViewId].
+  FlutterViewManager({int initialViewId = kImplicitViewId}) : _nextViewId = initialViewId;
+
+  // The counter from which viewIds are generated. Defaults to [kImplicitViewId].
+  int _nextViewId;
+  final Map<int, EngineFlutterView> _viewData = <int, EngineFlutterView>{};
+  final Map<int, JsFlutterViewOptions> _jsViewOptions = <int, JsFlutterViewOptions>{};
+
+  /// Exposes all the [EngineFlutterView]s registered so far.
+  Iterable<EngineFlutterView> get views => _viewData.values;
+
+  /// Retrieves an [EngineFlutterView] by its `viewId`.
+  EngineFlutterView? operator[](int viewId) {
+    return _viewData[viewId];
+  }
+
+  /// Creates (and stores) a view from its [JsFlutterViewOptions].
+  ///
+  /// Returns a `viewId` that is automatically assigned by this method.
+  int createView({ required JsFlutterViewOptions options }) {
+    assert(!_viewData.containsKey(_nextViewId)); // Adding the same view twice?
+
+    final int viewId = _nextViewId++;
+
+    _viewData[viewId] = EngineFlutterView.fromJsOptions(viewId, options);
+    _jsViewOptions[viewId] = options;
+
+    return viewId;
+  }
+
+  /// Returns the [JsFlutterViewOptions] associated to `viewId`.
+  ///
+  /// This is useful for plugins and apps that need this information, and can
+  /// be exposed through a method in ui_web.
+  JsFlutterViewOptions? getOptions(int viewId) {
+    return _jsViewOptions[viewId];
+  }
+
+  /// Unregisters [viewId].
```

Comment:
nit: consider expanding on what it means to unregister a viewId. For example, it is not clear whether the corresponding view is reusable after it is unregistered, and if so what can it do while it unregistered but not disposed of yet, as well as what needs to be done to dispose of it entirely.

Reply:
Yes, unregister is going to be more explicit than this (right now it's just "plumbing"). Discussed with Mouad that unregistering will also dispose of everything inside the view, thus making it unusable (unless it's re-registered)